monkey/common/credentials/lm_hash.py

```diff
@@ -1,16 +1,16 @@
 import re
 
-from pydantic import validator
-from pydantic.main import BaseModel
+from pydantic import SecretStr, validator
 
+from ..base_models import InfectionMonkeyBaseModel
 from .validators import ntlm_hash_regex
 
 
-class LMHash(BaseModel):
-    lm_hash: str
+class LMHash(InfectionMonkeyBaseModel):
+    lm_hash: SecretStr
 
     @validator("lm_hash")
-    def validate_hash_format(cls, nt_hash):
-        if not re.match(ntlm_hash_regex, nt_hash):
-            raise ValueError(f"Invalid LM hash provided: {nt_hash}")
-        return nt_hash
+    def validate_hash_format(cls, lm_hash):
+        if not re.match(ntlm_hash_regex, lm_hash.get_secret_value()):
```

Comment:
Let's do this (here, in NTHash, and wherever else relevant):
```suggestion
        if not re.match(ntlm_hash_regex, get_plain_text(lm_hash)):
```
It's consistent and we only have to change it in one place if required.

Reply:
Hmm, I'm not sure it's a good idea, since `get_plaintext` doesn't enforce the parameter to be a SecretStr. So if soemone accidentally starts using plaintext version of hashes/passwords nothing would break (maybe mypy would catch it, but not sure). Regarding one place to change it's this `get_secret_value` in secret object. If we migrate away from SecretStr, we could implement a class with `get_secret_value` and do whatever we want there.

Reply:
@mssalvatore